daemon/health.go

```diff
@@ -50,6 +52,37 @@ type probe interface {
 	run(context.Context, *Daemon, *container.Container) (*types.HealthcheckResult, error)
 }
 
+// httpProbe implements the "HTTP" probe type.
+type httpProbe struct {
+}
+
+func httpProbeRun(ctx context.Context, address, endpoint string, container *container.Container) (*types.HealthcheckResult, error) {
+	client := http.Client{
+		Timeout: timeoutWithDefault(container.Config.Healthcheck.Timeout, defaultProbeTimeout),
+	}
+	resp, err := client.Get(fmt.Sprintf("http://%s%s", address, endpoint))
```

Comment:
Just glancing over; guess we need to explicitly mention in the docs that `https://` is not supported